### PR TITLE
feat(r-grid): check the changes before fetching data

### PR DESCRIPTION
**Pull request by qqcky**
We need to check the changes before fetching data

### CODE REVIEW

File: packages/recomponents/src/components/r-repeater/r-repeater.vue [hunk 1]
```diff
@@ -28,6 +28,7 @@
 </template>
 
 <script>
+    import isEqual from 'lodash.isequal';
```
Comment (followyouhome) [inline packages/recomponents/src/components/r-repeater/r-repeater.vue]: we have local wrapper at packages/recomponents/src/common/helpers/is-equal.js

File: packages/recomponents/src/components/r-repeater/r-repeater.vue [hunk 2]
```diff
@@ -77,12 +78,14 @@
...
+        mounted() {
             this.fetchData();
         },
         watch: {
-            watcher() {
-                this.fetchData();
+            watcher(newValue, oldValue) {
+                if (isEqual(newValue, oldValue)) {
+                    this.fetchData();
+                }
```
Comment (followyouhome) [inline packages/recomponents/src/components/r-repeater/r-repeater.vue]: is it makes sense to add deep watch here?

Comment (followyouhome) [reply packages/recomponents/src/components/r-repeater/r-repeater.vue]: seems like does

Comment (qqcky) [reply packages/recomponents/src/components/r-repeater/r-repeater.vue]: Yes :) thank you!

**Pull request merged**